### DISCUSSION TITLE
"asdf" support?

### DISCUSSION THREAD

**Discussion by StanBright** [fullstaq-labs/fullstaq-ruby-server-edition#76] [General]
First, thanks for the great work with fulllstaq-ruby. It seems like it will benefit a lot of people.

I migrated all my servers to ASDF recently (away from RVM), and I'm wondering how difficult would it be to add ASDF support?

p.s. I run two relatively popular websites (~1M page-views/month), and I will be happy to blog about the results after moving to fullstaq-ruby.

**Comment by FooBarWidget** [comment]
If there is a lot of user demand then I'll consider it. So if anyone wants this, vote by adding a thumbs up emoticon.

W.r.t. effort: the main thing we need is the ability to register different Ruby versions on a global, system-wide basis, by dropping a file (or symlink) somewhere. We had to patch Rbenv to be able to do this. Rbenv normally looks for Ruby versions in ~/.rbenv/versions. This won't fly of course when combined with DEB/RPM packages. So we modified Rbenv to *also* look in /usr/lib/rbenv/versions. The Fullstaq Ruby packages then drop Ruby versions in /usr/lib/rbenv/versions.

**Comment by jamie** [comment]
I'm pretty sure asdf-ruby just leverages ruby-build, might not be much direct work needed for asdf.

**Reply by StanBright** [reply]
Yes, I just checked and `asdf-ruby` relies on `ruby-build`. However, I'm not sure what's the amount of work involved to get this working...

**Reply by FooBarWidget** [reply]
I think you're thinking about Fullstaq Ruby and asdf integration in a backward way.

The point of ruby-build is compilation. The point of Fullstaq Ruby is that we provide binaries, and you *don't* compile.

Therefore, integrating Fullstaq Ruby and asdf doesn't mean making ruby-build able to compile Fullstaq Ruby. It means that installing the Fullstaq Ruby packages automatically registers entries for asdf to use.

In other words: Fullstaq Ruby replaces the need for ruby-build.

I took a quick look at [asdf-ruby](https://github.com/asdf-vm/asdf-ruby) and it looks like it's tightly coupled with ruby-build. I think that Fullstaq Ruby integration with asdf would require a new asdf plugin that simply looks for precompiled binaries in /usr/lib/fullstaq-ruby/versions, instead of shelling out to ruby-build.

**Reply by jwarner112** [reply]
Having learned about Fullstaq-Ruby and loving the idea, and using ASDF myself, I quickly found this discussion and promptly filed [this issue](https://github.com/asdf-vm/asdf-ruby/issues/275) on their repo. If I get decent feedback and time to work on it, I will.
